solvis/solution/inversion_solution/inversion_solution_file.py

```diff
@@ -237,48 +273,82 @@ def get_regime() -> str:
                 f"expected Fault Regime missing in solution logic tree, see {self.LOGIC_TREE_PATH}."
             )  # pragma: no cover
 
-        if not self._fault_regime:
-            self._fault_regime = get_regime()
-        return self._fault_regime
+        return get_regime()
 
     @property
     def rupture_rates(self) -> 'DataFrame[RuptureRateSchema]':
-        dtypes: defaultdict = defaultdict(lambda: 'Float32')
-        # dtypes = {}
-        dtypes["Rupture Index"] = 'UInt32'  # pd.UInt32Dtype()
-        dtypes["fault_system"] = pd.CategoricalDtype()
-        # dtypes["Annual Rate"] = 'Float32' # pd.Float32Dtype()
-        # return pd.read_csv(zipfile.Path(self._archive_path, at=self.RATES_PATH).open(), dtype=dtypes)
-        df = self._dataframe_from_csv(self._rates, self.RATES_PATH, dtypes)
-        return cast('DataFrame[RuptureRateSchema]', df)
+        """
+        Get the rupture rates from the archive.
+
+        Returns:
+            A DataFrame containing rupture rate data.
+        """
+        if self._rates is None:
+            dtypes: defaultdict = defaultdict(lambda: 'Float32')
+            dtypes["Rupture Index"] = 'UInt32'  # pd.UInt32Dtype()
+            dtypes["fault_system"] = pd.CategoricalDtype()
+            self._rates = self._dataframe_from_csv(self.RATES_PATH, dtypes)
+        return cast('DataFrame[RuptureRateSchema]', self._rates)
 
     @property
+    @cache
     def ruptures(self) -> 'DataFrame[RuptureSchema]':
-        dtypes: defaultdict = defaultdict(lambda: 'Float32')
-        # dtypes = {}
-        dtypes["Rupture Index"] = 'UInt32'
-        df = self._dataframe_from_csv(self._ruptures, self.RUPTS_PATH, dtypes)
-        return cast('DataFrame[RuptureSchema]', df)
+        """
+        Get the ruptures from the archive.
+
+        Returns:
+            A DataFrame containing rupture data.
+        """
+        if self._ruptures is None:
```

Comment:
is this test necessary if using the `@cache` decorator?

Reply:
A better approach might be a  InversionSolution.from_props() static method that returns the new solution. This would replace the `__init__()  ... set_props() ` dance.

Reply:
ref #76